src/index.js

```diff
@@ -21,19 +22,21 @@ root.render(
    <React.StrictMode>
       <BrowserRouter>
          <AuthProvider>
-            <PlaylistProvider>
-               <HistoryProvider>
-                  <WatchlaterProvider>
-                     <LikeProvider>
-                        <VideoProvider>
-                           <IconProvider>
-                              <App />
-                           </IconProvider>
-                        </VideoProvider>
-                     </LikeProvider>
-                  </WatchlaterProvider>
-               </HistoryProvider>
-            </PlaylistProvider>
+            <ThemeProvider>
+               <PlaylistProvider>
+                  <HistoryProvider>
+                     <WatchlaterProvider>
+                        <LikeProvider>
+                           <VideoProvider>
+                              <IconProvider>
+                                 <App />
+                              </IconProvider>
+                           </VideoProvider>
+                        </LikeProvider>
+                     </WatchlaterProvider>
+                  </HistoryProvider>
+               </PlaylistProvider>
+            </ThemeProvider>
          </AuthProvider>
       </BrowserRouter>
```

Comment:
keep Auth Provider at the top  of files because sometime later you may need to Private routes so in that case you will need auth and may even end up getting errors without realizing the actual issue. I would suggest to do these changes in this PR itself before merging

Reply:
Hey Bhushan, the authprovider is just below the browserprovider(which is provided by react router), and above all the other providers. I have implemented privates routes and it's working fine in this way.